### PR TITLE
Refactor bubble open events

### PR DESCRIPTION
**Pull request by moniika**
## The basics

- [ ] I branched from develop
- [ ] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Resolves

Part of #4203

### Proposed Changes

Creates dedicated class (`Blockly.Events.BubbleOpen`) for bubble open events.

### Reason for Changes

See issue, but generally motivated by:
- better typing
- clearer events

### Test Coverage

Tested by logging events on playground and observing that expected events were logged.

Tested on:
 * Desktop Chrome

### CODE REVIEW

File: core/events_bubble_open.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+   */
+  this.element = opt_element;
+};
+Blockly.utils.object.inherits(Blockly.Events.BubbleOpen, Blockly.Events.Ui);
+
+/**
+ * Type of this event.
+ * @type {string}
+ */
+Blockly.Events.BubbleOpen.prototype.type = Blockly.Events.BUBBLE_OPEN;
```
Comment (samelhusseini) [inline core/events_bubble_open.js:53]: Why are we inheriting from UI events, if we're overriding the type. 

If someone wants to filter for all events that are not UI events, then they won't be able to just do this no more: 
```
if (event.type !== 'ui') {
 ...
}
```

File: core/events_bubble_open.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Events fired as a result of bubble open.
+ * @author kozbial@google.com (Monica Kozbial)
+ */
+'use strict';
+
+goog.provide('Blockly.Events.BubbleOpen');
```
Comment (samelhusseini) [inline core/events_bubble_open.js:13]: We could consider moving all these events into their own folder for better organization.

**Review by samelhusseini — COMMENTED** [review]
Any reason to check in the _compressed files?

File: core/events_bubble_open.js (hunk 2 above)
Comment (moniika) [reply core/events_bubble_open.js:13]: Yes, I was planning on moving the files after I get most PRs in.

**Comment by moniika** [comment]
> Any reason to check in the _compressed files?

I need to rebuild because I added new files and edited provide/requires

File: core/events_bubble_open.js (hunk 1 above)
Comment (moniika) [reply core/events_bubble_open.js:53]: UI events have a new property `IS_UI_EVENT`, in order to filter UI events, one can:
```
if (event.IS_UI_EVENT) {
  ...
}
```

Comment (samelhusseini) [reply core/events_bubble_open.js:53]: Ah got it.

**Review by samelhusseini — APPROVED** [review]

**Pull request merged**